### PR TITLE
Upgrade libzmq 4.2.0

### PR DESCRIPTION
**Pull request by lgeiger**
This also simplifies the build script and move the essential stuff to cross platform js.
It will  make it possible to build libzmq from source on windows if someone writes a batch script handling the build.

### CODE REVIEW

File: binding.gyp [hunk 1]
```diff
@@ -14,6 +14,7 @@
           'libraries': [
             '<(PRODUCT_DIR)/../../windows/lib/libzmq',
             'ws2_32.lib',
+            'iphlpapi'
```
Comment (rgbkrk) [inline binding.gyp]: Given I have no background on this, it looks like someone fell asleep on the keyboard to name this.

**Comment by rgbkrk** [comment]
Im glad this leaves room for building from source on Windows.

File: binding.gyp (hunk 1 above)
Comment (willingc) [inline binding.gyp]: lol. IP Helper API. https://msdn.microsoft.com/en-us/library/windows/desktop/aa366073(v=vs.85).aspx

A good example of how adding an 'e' to make a human readable name would have been a solid idea. `iphelpapi` would have been better than `iphlpapi`.

Overall, I think the addition here is fine.

**Comment by willingc** [comment]
@lgeiger Wow! Lots of good work here. I didn't do a line-by-line review rather a quick scan. I would say merge it.

**Comment by lgeiger** [comment]
@willingc `zmq.h` and `zmq_utils.h` are just copied from the latest release of libzmq so nothing interesting there 😄 .

**Comment by willingc** [comment]
💯 to humility @lgeiger.

**Pull request merged**